examples/piecestore/rpc/client/api/api.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package api
+
+import (
+	"errors"
+	"io"
+	"log"
+
+	"github.com/zeebo/errs"
+
+	"golang.org/x/net/context"
+
+	pb "storj.io/storj/protos/piecestore"
+)
+
+var serverError = errs.Class("serverError")
+
+// PieceMeta -- Struct containing Piece information from PieceMetaRequest
+type PieceMeta struct {
+	ID         string
+	Size       int64
+	Expiration int64
+}
+
+// PieceMetaRequest -- Request info about a piece by Id
+func PieceMetaRequest(ctx context.Context, c pb.PieceStoreRoutesClient, id string) (*PieceMeta, error) {
+
+	reply, err := c.Piece(ctx, &pb.PieceId{Id: id})
+	if err != nil {
+		return nil, err
+	}
+
+	return &PieceMeta{ID: reply.Id, Size: reply.Size, Expiration: reply.Expiration}, nil
+}
+
+// StorePieceRequest -- Upload Piece to Server
+func StorePieceRequest(ctx context.Context, c pb.PieceStoreRoutesClient, id string, data io.Reader, dataOffset int64, length int64, ttl int64, storeOffset int64) error {
+	stream, err := c.Store(ctx)
```

Comment:
should there be something deferred for closing the stream in the case that this function exits unsuccessfully?

Reply:
👍